dash/dash-renderer/src/components/error/menu/VersionInfo.react.js

```diff
@@ -0,0 +1,160 @@
+    const lastDismissedVersion = localStorage.getItem('lastDismissedVersion');
+    if (
+        currentDashVersion == newDashVersion ||
+        !showNotifications ||
+        newDashVersion === undefined
+    ) {
+        return false;
+    } else if (
+        lastDismissed &&
+        Date.now() - Number(lastDismissed) > DAY_IN_MS
+    ) {
+        return true;
+    } else if (
+        lastDismissedVersion &&
+        !lastDismissed &&
+        compareVersions(
+            localStorage.getItem('lastDismissedVersion'),
+            newDashVersion
+        ) < 0
+    ) {
+        return true;
+    } else {
+        return !lastDismissed && !lastDismissedVersion;
+    }
+}
+
+export const VersionInfo = ({
+    config,
+    showNotifications,
+    setShowNotifications
+}) => {
+    const [upgradeInfo, setUpgradeInfo] = useState([]);
+    const [upgradeTooltipOpened, setUpgradeTooltipOpened] = useState(false);
+
+    const newDashVersion = upgradeInfo[0] ? upgradeInfo[0].version : undefined;
+
+    const setDontShowAgain = () => {
+        // Set local storage to record the last dismissed notification
+        setUpgradeTooltipOpened(false);
+        setShowNotifications(false);
+    };
+
+    const setRemindMeLater = () => {
+        // Set local storage to record the last dismissed notification
+        localStorage.setItem('lastDismissed', Date.now());
+        setUpgradeTooltipOpened(false);
+    };
+
+    const setSkipThisVersion = () => {
+        // Set local storage to record the last dismissed version
+        localStorage.setItem('lastDismissedVersion', newDashVersion);
+        setUpgradeTooltipOpened(false);
+    };
+
+    useEffect(() => {
+        if (showNotifications) {
+            requestDashVersionInfo(
+                config.dash_version,
+                config.dash_version_url
+            ).then(body => {
+                setUpgradeInfo(body);
+            });
+        }
+    }, [showNotifications]);
+
+    useEffect(() => {
+        const hideUpgradeTooltip = e => {
+            if (
+                upgradeTooltipOpened &&
+                !e.target.matches(
+                    '.dash-debug-menu__version, .dash-debug-menu__version *'
+                )
+            ) {
+                setUpgradeTooltipOpened(false);
+            }
+        };
+        // Close the upgrade tooltip if the user clicks outside of it
+        document.addEventListener('click', hideUpgradeTooltip);
+
+        return () => document.removeEventListener('click', hideUpgradeTooltip);
+    }, [upgradeTooltipOpened]);
+
+    return (
+        <div className='dash-debug-menu__version'>
+            {upgradeTooltipOpened ? (
+                <div className='dash-debug-menu__upgrade-tooltip'>
+                    <a
+                        target='_blank'
+                        href='https://dash.plotly.com/installation'
+                    >
+                        Read details
```

Comment:
Is this intended to link to detail on how to install it or details about the features in the version?

Reply:
Yeah! I didn't have anywhere to point to yet but I should definitely add that to the TODO's.